tests/test_chmod_command.rs

```diff
@@ -0,0 +1,192 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::path::Path;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+fn create_commit(
+    test_env: &TestEnvironment,
+    repo_path: &Path,
+    name: &str,
+    parents: &[&str],
+    files: &[(&str, &str)],
+) {
+    if parents.is_empty() {
+        test_env.jj_cmd_success(repo_path, &["new", "root", "-m", name]);
+    } else {
+        let mut args = vec!["new", "-m", name];
+        args.extend(parents);
+        test_env.jj_cmd_success(repo_path, &args);
+    }
+    for (name, content) in files {
+        std::fs::write(repo_path.join(name), content).unwrap();
+    }
+    test_env.jj_cmd_success(repo_path, &["branch", "create", name]);
+}
+
+fn get_log_output(test_env: &TestEnvironment, repo_path: &Path) -> String {
+    test_env.jj_cmd_success(repo_path, &["log", "-T", "branches"])
+}
+
+#[test]
+fn test_chmod_simple_conflict() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    create_commit(&test_env, &repo_path, "base", &[], &[("file", "base\n")]);
+    create_commit(&test_env, &repo_path, "n", &["base"], &[("file", "n\n")]);
+    create_commit(&test_env, &repo_path, "x", &["base"], &[("file", "x\n")]);
+    // Test chmodding a file. The effect will be visible in the conflict below.
+    test_env.jj_cmd_success(&repo_path, &["chmod", "x", "file", "-r=x"]);
+    create_commit(&test_env, &repo_path, "conflict", &["x", "n"], &[]);
+    // Test the setup
+    insta::assert_snapshot!(get_log_output(&test_env, &repo_path), @r###"
+    @    conflict
+    ├─╮
+    ◉ │  x
+    │ ◉  n
+    ├─╯
+    ◉  base
+    ◉
+    "###);
+
+    let stdout = test_env.jj_cmd_success(&repo_path, &["cat", "file"]);
+    insta::assert_snapshot!(stdout, 
+    @r###"
+    Conflict:
+      Removing file with id df967b96a579e45a18b8251732d16804b2e56a55
+      Adding executable file with id 587be6b4c3f93f93c489c0111bba5596147a26cb
+      Adding file with id 8ba3a16384aacc37d01564b28401755ce8053f51
+    "###);
+    test_env.jj_cmd_success(&repo_path, &["chmod", "x", "file"]);
+    let stdout = test_env.jj_cmd_success(&repo_path, &["cat", "file"]);
+    insta::assert_snapshot!(stdout, 
+    @r###"
+    Conflict:
+      Removing executable file with id df967b96a579e45a18b8251732d16804b2e56a55
+      Adding executable file with id 587be6b4c3f93f93c489c0111bba5596147a26cb
+      Adding executable file with id 8ba3a16384aacc37d01564b28401755ce8053f51
+    "###);
+
+    // An error doesn't prevent `chmod` to change the files it can.
+    // In this case, it successfully makes `file` normal and fails with
+    // `nonexistent`.
+    let stderr = test_env.jj_cmd_failure(&repo_path, &["chmod", "n", "nonexistent", "file"]);
+    insta::assert_snapshot!(stderr, @r###"
```

Comment:
I realized this past of the test doesn't show what I thought it did, and the behavior should be the opposite of what I said. I'll fix it.